lib/ballot.go

```diff
@@ -400,6 +413,15 @@ func (b *BallotBoxes) AddBallot(ballot Ballot) (isNew bool, err error) {
 	return
 }
 
+func (b *BallotBoxes) AddSource(m sebakcommon.Message) {
+	b.Sources[m.Source()] = m.GetHash()
```

Comment:
@spikeekips I have a question.
`b.Sources` save only one recently received message from the `source`, right?